crates/phactory/src/contracts/pink.rs

```diff
@@ -112,7 +115,35 @@ impl contracts::NativeContract for Pink {
                 if ink_result.result.is_err() {
                     log::error!("Pink [{:?}] query exec error: {:?}", self.id(), ink_result);
                 }
-                return Ok(Response::InkMessageReturn(ink_result.encode()));
+                return Ok(Response::Payload(ink_result.encode()));
+            }
+            Query::SidevmQuery(payload) => {
+                let handle = context
+                    .sidevm_handle
+                    .as_ref()
+                    .ok_or(QueryError::SidevmNotFound)?;
+                let cmd_sender = match handle {
+                    contracts::SidevmHandle::Terminated(_) => {
+                        return Err(QueryError::SidevmNotFound)
+                    }
+                    contracts::SidevmHandle::Running(sender) => sender,
+                };
+                let origin = origin.clone().into();
```

Comment:
Looks like the origin is always required for ink. However, it is not signed in case of calling from RPC, So it can be set to an arbitrary account.
Allow being None for sidevm sounds reasonable.